### PR TITLE
Created first draft of semver post

### PR DESCRIPTION
**Pull request by hannesvdvreken**
### Added

* Post about tilde and caret operators and shameless self-promotion for semver.mwl.be

### CODE REVIEW

File: _posts/2015-04-10-tilde-and-caret-constraints.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+---
+layout: post
+title:  "Tilde and caret version constraints in Composer"
+date:   2015-04-10
+author: hannesvdvreken
+categories: [composer]
+---
+
+When installing a dependency we are used to install a version of the depedency that adheres to a
```
Comment (bramdevries) [inline _posts/2015-04-10-tilde-and-caret-constraints.md]: second `dependency` is spelled wrong, but also might be used one to many times im the opening sentence

File: _posts/2015-04-10-tilde-and-caret-constraints.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+when you upgrade to a new version. Basically when the `x` in a `x.y.z` version number changes,
+you might need to do some changes to be able to work with this new version without problems.
+When dealing with a `0.y.z` version this is slightly different: when the first non-zero number (`y`) changes,
+backwards compatibility is not guaranteed. If you're not familiar with semver:
+head over to the [specs](http://semver.org/)!
+
+A dependency can change without breaking backwards compatibility. A new release will then have a
+greater version number (x.y), depending on the type of change (bugfix or added functionality).
+When you depend on packages you might want to define a range of versions that are OK to install,
+rather than having a locked down version specified. This allows you to get receive fixes
```
Comment (bramdevries) [inline _posts/2015-04-10-tilde-and-caret-constraints.md]: get receive should be just receive

File: _posts/2015-04-10-tilde-and-caret-constraints.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+(that includes security fixes) from external code for free with a simple update command.
```
Comment (bramdevries) [inline _posts/2015-04-10-tilde-and-caret-constraints.md]: that also includes security fixes

File: _posts/2015-04-10-tilde-and-caret-constraints.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+Depending on your dependency manager you can define version contraints using wildcards (`*`), comparators like `<=`,
+logical operators (`,` often means AND and `|` means OR), etc... Using logical operators you can mix and match
+different version constraints to build even more complex version constraints. For example `>=0.9.0,<0.11.0`
```
Comment (bramdevries) [inline _posts/2015-04-10-tilde-and-caret-constraints.md]: 2x version contraints in the same sentence

File: _posts/2015-04-10-tilde-and-caret-constraints.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+There are also some syntactic sugar operators like `~` (tilde) and `^` (caret).
+
+- [`~4.1.3`](http://semver.mwl.be/#?package=laravel%2Fframework&version=~4.2.3) means `>=4.1.3,<4.2.0`,
+- [`~4.1`](http://semver.mwl.be/#?package=laravel%2Fframework&version=~4.2) means `>=4.1.0,<5.0.0` (most used) and
+- [`~4`](http://semver.mwl.be/#?package=laravel%2Fframework&version=~4.2.3) means `>=4.0.0,<5.0.0`.
+
+The caret sign is slightly different:
+
+- [`^4.1.3`](http://semver.mwl.be/#?package=laravel%2Fframework&version=%5E4.1.3) (most used) means `>=4.1.3,<5.0.0`,
+- [`^4.1`](http://semver.mwl.be/#?package=laravel%2Fframework&version=%5E4.1) means `>=4.1.0,<5.0.0` and
```
Comment (bramdevries) [inline _posts/2015-04-10-tilde-and-caret-constraints.md]: How is this different than using a `~4.1` ?

File: _posts/2015-04-10-tilde-and-caret-constraints.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+- [`^4`](http://semver.mwl.be/#?package=laravel%2Fframework&version=%5E4) means `>=4.0.0,<5.0.0`
+which is the same as `~4` and `4.*`.
```
Comment (bramdevries) [inline _posts/2015-04-10-tilde-and-caret-constraints.md]: Same here it's not different than `~4`

File: _posts/2015-04-10-tilde-and-caret-constraints.md (hunk 5 above)
Comment (hannesvdvreken) [inline _posts/2015-04-10-tilde-and-caret-constraints.md]: it's not different, actually. I'll add examples with 0.y.z to show the difference.

**Comment by hannesvdvreken** [comment]
thanks for the feedback @bramdevries!

File: _posts/2015-04-10-tilde-and-caret-constraints.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+you will be able to exclude all earlier versions with
+[`^2.1.1`](http://semver.mwl.be/#?package=league%2Fevent&version=%5E2.1.1). 
+This way you can still benefit from all new backward compatible releases.
+
+Fun fact: `caret` is Latin for "it lacks", so you can remember this as "it (this version range) lacks
+some early releases but it's backwards compatible up to the next major release".
+
+For now when you run the command `composer require monolog/monolog` it will still save `~1.13`
+(at the time of writing) as the default version constraint in your composer.json file and not `^1.13.1`.
+This is to protect users who might not have update their composer to a version which can handle
```
Comment (driesvints) [inline _posts/2015-04-10-tilde-and-caret-constraints.md]: not have update**d**

Comment (driesvints) [inline _posts/2015-04-10-tilde-and-caret-constraints.md]: Composer is a name so should always be written with a capital letter.

**Comment by hannesvdvreken** [comment]
thanks for the feedback @driesvints!

File: _posts/2015-04-10-tilde-and-caret-constraints.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+you will be able to exclude all earlier versions with
+[`^2.1.1`](http://semver.mwl.be/#?package=league%2Fevent&version=%5E2.1.1). 
+This way you can still benefit from all new backward compatible releases.
+
+Fun fact: `caret` is Latin for "it lacks", so you can remember this as "it (this version range) lacks
+some early releases but it's backwards compatible up to the next major release".
+
+For now when you run the command `composer require monolog/monolog` it will still save `~1.13`
+(at the time of writing) as the default version constraint in your composer.json file and not `^1.13.1`.
+This is to protect users who might not have updated their Composer to a version which can handle
```
Comment (driesvints) [inline _posts/2015-04-10-tilde-and-caret-constraints.md]: This reads a little weird. 

who might not have updated their Composer **dependencies** 

might be a little better.

Comment (hannesvdvreken) [inline _posts/2015-04-10-tilde-and-caret-constraints.md]: I meant

"updated their Composer *installation*"

**Pull request merged**